corehq/apps/hqcase/case_helper.py

```diff
@@ -163,6 +168,128 @@ def close(
             is_creation=False,
         )
 
+    @classmethod
+    def copy_cases(
+        cls,
+        domain,
+        case_ids,
+        to_owner,
+        censor_data=None,
+        count_only=False
+    ):
+        """
+        Copies the cases governed by 'case_ids' to the respective 'to_owner'
+        and replacing the relevant case properties with the specified 'replace_props'
+        on the copied cases (useful for hiding sensitive data).
+
+        :param domain: the domain string
+        :param case_ids: case ids of the cases to copy
+        :param to_owner: the owner to copy the cases to
+        :param censor_data: the attributes/properties to be censored, specified as a dict.
+                The keys corresponding to the property/attribute and the value specifies
+                the type of censored data, i.e. number or date
+        :param count_only: specifies whether to return only the number of copied cases.
+
+        :return: The copied cases. If `count_only` is True only the count will be returned.
+        """
+        from corehq.apps.hqcase.utils import CASEBLOCK_CHUNKSIZE
+
+        if not to_owner:
+            raise Exception('Must copy cases to valid new owner')
+
+        original_cases = CommCareCase.objects.get_cases(case_ids=case_ids, domain=domain)
+        if not any(original_cases):
+            return [] if not count_only else 0
+
+        processed_cases = {}
+        copied_cases_case_blocks = []
+
+        def _get_new_identifier_index(case_identifier_index):
```

Comment:
Toughts on moving these methods outside the function?

Reply:
@SmittieC I created it as nested functions because the functions are by nature very coupled to the copy method and is not useful outside of the copy method. Other than that I don't have strong sentiments for keeping it nested. I wonder what others think?